web/src/components/RemoteContent.tsx

```diff
@@ -59,68 +60,46 @@ const SandBox = styled.div<{ centered: boolean }>`
 `
 
 type PropsType = {
-  dangerouslySetInnerHTML: {
-    __html: string
-  }
+  html: string
   onInternalLinkClick: (url: string) => void
-  centered: boolean
+  centered?: boolean
 }
 
 const HIJACK = new RegExp(buildConfig().internalLinksHijackPattern)
 
-class RemoteContent extends React.Component<PropsType> {
-  static defaultProps = {
-    centered: false
-  }
-
-  sandBoxRef: RefObject<HTMLDivElement>
+const RemoteContent = ({ html, onInternalLinkClick, centered = false }: PropsType): ReactElement => {
+  const sandBoxRef: RefObject<HTMLDivElement> = React.createRef<HTMLDivElement>()
 
-  handleClick = (event: MouseEvent): void => {
-    // https://stackoverflow.com/a/1000606
-    // $FlowFixMe
-    event.preventDefault ? event.preventDefault() : (event.returnValue = false)
-    const target: EventTarget | null = event.currentTarget
-
-    if (target instanceof HTMLAnchorElement) {
-      const href = target.href
-      this.props.onInternalLinkClick(decodeURIComponent(new URL(decodeURIComponent(href)).pathname))
-    }
-  }
+  const handleClick = useCallback(
+    (event: MouseEvent): void => {
+      event.preventDefault ? event.preventDefault() : (event.returnValue = false)
+      const target: EventTarget | null = event.currentTarget
 
-  constructor(props: PropsType) {
-    super(props)
-    this.sandBoxRef = React.createRef<HTMLDivElement>()
-  }
+      if (target instanceof HTMLAnchorElement) {
+        const href = target.href
+        onInternalLinkClick(decodeURIComponent(new URL(decodeURIComponent(href)).pathname))
+      }
+    },
+    [onInternalLinkClick]
+  )
 
-  hijackATags(): void {
-    if (!this.sandBoxRef.current) {
+  useEffect(() => {
+    if (!sandBoxRef.current) {
       return
     }
-    const collection: HTMLCollectionOf<HTMLAnchorElement> = this.sandBoxRef.current.getElementsByTagName('a')
+    const collection: HTMLCollectionOf<HTMLAnchorElement> = sandBoxRef.current.getElementsByTagName('a')
     Array.from(collection).forEach(node => {
       if (HIJACK.test(node.href)) {
-        node.addEventListener('click', this.handleClick)
+        node.addEventListener('click', handleClick)
       }
     })
-  }
+  }, [html, handleClick, sandBoxRef])
 
-  componentDidMount(): void {
-    this.hijackATags()
+  const dangerouslySetInnerHTML = {
+    __html: Dompurify.sanitize(html)
   }
```

Comment:
The important change in this PR :)

Reply:
So I guess this means we discarded the idea to sanitize the content in the api client? Or do I mix this up with the news?

Reply:
Yes, the problem was that neither dompurify nor isomorphic-dompurify work on native (isomorphic-dompurify only works in browser and node). Also imo it is better to sanitize where it is potentially dangerous, i.e. when setting it as inner html. That way, we make sure everything is really sanitized and safe.